code/modules/mob/living/carbon/human/species/vox.dm

```diff
@@ -84,16 +84,21 @@
 /datum/species/vox/after_equip_job(datum/job/J, mob/living/carbon/human/H)
 	if(!H.mind || !H.mind.assigned_role || H.mind.assigned_role != "Clown" && H.mind.assigned_role != "Mime")
 		H.unEquip(H.wear_mask)
-	H.unEquip(H.l_hand)
 
 	H.equip_or_collect(new /obj/item/clothing/mask/breath/vox(H), slot_wear_mask)
 	var/tank_pref = H.client && H.client.prefs ? H.client.prefs.speciesprefs : null
+	/var/obj/item/tank/internal_tank
 	if(tank_pref)//Diseasel, here you go
-		H.equip_or_collect(new /obj/item/tank/nitrogen(H), slot_l_hand)
+		internal_tank = new /obj/item/tank/nitrogen(H)
 	else
-		H.equip_or_collect(new /obj/item/tank/emergency_oxygen/vox(H), slot_l_hand)
-	to_chat(H, "<span class='notice'>You are now running on nitrogen internals from the [H.l_hand] in your hand. Your species finds oxygen toxic, so you must breathe nitrogen only.</span>")
-	H.internal = H.l_hand
+		internal_tank = new /obj/item/tank/emergency_oxygen/vox(H)
+	if (!H.equip_to_appropriate_slot(internal_tank))
+		if (!H.put_in_any_hand_if_possible(internal_tank))
+			H.unEquip(H.l_hand)
+			H.equip_or_collect(internal_tank, slot_l_hand)
+			to_chat(H, "<span class='boldannounce'>Could not find an empty slot for internals! Please report this as a bug</span>")
+	internal_tank.toggle_internals(H)
```

Comment:
What happens at this line if the code at lines 97-99 was executed?
Do they end up with a running tank, but the tank is embedded in their torso and inaccessible?
Or does this runtime since internal_tank doesn't exist?

Just curious.